### DISCUSSION TITLE
Automatically collect prompt eval duration

### DISCUSSION THREAD

**Discussion by Ambrevar** [joaotavora/sly#414] [Ideas]
Wouldn't be nice to automatically collect the duration of a prompt?
I'm doing this in a customized prompt, but maybe SLY should do this by default?
Everything is already in place, we just need to store the duration in the text property of the result, just like we store the back-reference.

Thoughts?
I can send a patch if you like! :)

**Comment by joaotavora** [comment]
Hmm, i think it's a pretty nice idea.  Maybe we could even show it by default, maybe.  But this has to be done on the Lisp side, and the time collected as precisely as possible.  Maybe only collect the start time when the user's expression is actually evaluated. Or maybe the end-time too.  Then it should be sent back to Emacs which can do whatever fanciness it want with the raw data.

Do send a patch yes.

**Comment by joaotavora** [comment]
By "whatever fanciness" I mean show it if it's more than a predetermined amount of time, say 2 seconds.  Maybe even the default prompt could do that.  But please leave this part of later, focus on designing the collect and transmission of the data.

**Comment by Ambrevar** [comment]
João Távora <notifications@github.com> writes:

> By "whatever fanciness" I mean show it if it's more than a predetermined amount of time, say 2 seconds.  Maybe even the default prompt could do that.  But please leave this part of later, focus on designing the collect and transmission of the data.

Yes, this is exactly what I've implemented for my FOSDEM presentation or
here in this article:

https://ambrevar.xyz/lisp-repl-shell/index.html#orga2d366f

I've implemented it on the Emacs side only: it's less precise, but my
need were only to a precision <100ms.

I'll see what I can come up with for the Lisp side of things.

**Reply by joaotavora** [reply]
Yes, I read your article.  Congratulations

**Comment by Ambrevar** [comment]
By the way, for ideal results this would need the prompt coloring to be
fixed.

Should I open an issue about it so we don't forget?

**Reply by joaotavora** [reply]
> Should I open an issue about it so we don't forget?

 :-) Hmmm I think conversation is good enough for now.

**Comment by Ambrevar** [comment]
Thanks! :)
